consensus/polybft/block_builder.go

```diff
@@ -82,31 +61,37 @@ type BlockBuilder struct {
 	state *state.Transition
 }
 
-// Reset is used to indicate that the current block building has been interrupted
-// and it has to clean any data
-func (b *BlockBuilder) Reset() error {
+// Init initializes block builder before adding transactions and actual block building
+func (b *BlockBuilder) Init() error {
+	// set the timestamp
+	parentTime := time.Unix(int64(b.params.Parent.Timestamp), 0)
```

Comment:
Can we check what other chains do with this? I am not sure what is the consensus, should the timestamp be the current one or the time when the block was meant to be created?

Reply:
This is just copy pasted code as it was before. The only "subtle" difference is that timestamp was set after all transactions have been written and now it is before

Reply:
The full logic is this (as it was before):
1. take the timestamp from the parent
2. add block time
3. if that timestamp is lower than current time update it to current time (` time.Now()`)
4. write that timestamp to header  as uint64 (`uint64(headerTime.Unix())`)

Reply:
I think (if I understand correctly - I was looking at `miner.go` in go-eth), this is how setting timestamp working in `go-eth`:
1. Before transactions are filed the timestamp is created as:
```
	timestamp := genParams.timestamp
	if parent.Time() >= timestamp {
		if genParams.forceTime {
			return nil, fmt.Errorf("invalid timestamp, parent %d given %d", parent.Time(), timestamp)
		}
		timestamp = parent.Time() + 1
	}
```
2. `genParams.timestamp` is set to `time.Now().Unix()`

I think we are good to go with our timestamp setting